applications/plugins/Communication/components/serverCommunicationVRPN.inl

```diff
@@ -0,0 +1,399 @@
+        }
+    }
+}
+
+void VRPN_CALLBACK ServerCommunicationVRPN::processAnalogMessage(void *userdata, const vrpn_ANALOGCB a)
+{
+    int nbChannels = a.num_channel;
+
+    ServerCommunicationVRPN* instance = static_cast<ServerCommunicationVRPN*>(userdata);
+    std::map<std::string, CommunicationSubscriber*> subscribersMap = instance->getSubscribers();
+
+    ArgumentList analogStream;
+
+    if (a.num_channel>1)
+    {
+        int row=0, col=0;
+        try
+        {   // Matrix will have a single row but the number of columns depends on number of channels
+            row = 1;
+            col = a.num_channel;
+            if (row < 0 || col < 0)
+                return;
+        }
+        catch(std::invalid_argument& e)
+        {
+            msg_warning(instance) << "no available conversion for: " << e.what();
+            return;
+        }
+        catch(std::out_of_range& e)
+        {
+            msg_warning(instance) << "out of range : " << e.what();
+            return;
+        }
+
+        for (int i = 0; i < a.num_channel; i++)
+        {
+            std::string stream = "VRPNdouble:";
+            stream.append(std::to_string(a.channel[i]));
+            analogStream.push_back(stream);
+        }
+
+        if(analogStream.size() == 0)
+        {
+            msg_error(instance) << "argument list size is empty";
+            return;
+        }
+
+        if((unsigned int)row*col != analogStream.size())
+        {
+            msg_error(instance) << "argument list size is != row/cols; " << analogStream.size() << " instead of " << row*col;
+            return;
+        }
+
+        for (std::map<std::string, CommunicationSubscriber*>::iterator it = subscribersMap.begin(); it != subscribersMap.end(); it++)
+        {
+            CommunicationSubscriber* subscriber = it->second;
+            instance->saveDatasToReceivedBuffer(subscriber->getSubject(), analogStream, row, col);
+        }
+    }
+
+    else
+    {
+        std::string stream = "VRPNdouble:";
+        stream.append(std::to_string(a.channel[0]));
+        analogStream.push_back(stream);
+        std::cout << stream << std::endl;
+
+        for (std::map<std::string, CommunicationSubscriber*>::iterator it = subscribersMap.begin(); it != subscribersMap.end(); it++)
+        {
+            CommunicationSubscriber* subscriber = it->second;
+            instance->saveDatasToReceivedBuffer(subscriber->getSubject(), analogStream, -1, -1);
+        }
+    }
+}
+
+void VRPN_CALLBACK ServerCommunicationVRPN::processTrackerMessage(void *userdata, const vrpn_TRACKERCB z)
+{
+    ServerCommunicationVRPN* instance = static_cast<ServerCommunicationVRPN*>(userdata);
+    std::map<std::string, CommunicationSubscriber*> subscribersMap = instance->getSubscribers();
+    ArgumentList trackerStream;
+    int row, col;
+
+    try
+    {   // Matrix will have a single row but the number of columns will be 3.
```

Comment:
?

Reply:
Because the pos is an array of 3 inside the scene. I couldn't think of a better comment. 😉

Reply:
Tracker can be more than 3.